src/prefect/agent/docker/agent.py

```diff
@@ -398,11 +398,18 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
         # Create a container
         self.logger.debug("Creating Docker container {}".format(image))
 
-        host_config = {
-            "auto_remove": True,
+        # By default, auto-remove containers
+        host_config: Dict[str, Any] = {"auto_remove": True}
+
+        docker_engine_version = tuple(
+            int(i) for i in self.docker_client.version()["Version"].split(".")
+        )
+        if docker_engine_version >= (20, 10, 0):
```

Comment:
For this version check I recommend using 
```python
import packaging

if packaging.version.parse(self.docker_client.version()["Version"]) < packaging.version.parse("20.10.0"):
```
and emphasizing / conditioning on the "bad version" instead of the good one.

Reply:
Whoa packaging.version.parse 😮 I was mostly concerned about my bad parser so this looks great.